src/pymor/algorithms/phdmd.py

```diff
@@ -0,0 +1,474 @@
+
+    The formal description of this algorithm can be found in :cite:`MMU22`.
+
+    Parameters
+    ----------
+    X
+        The `NumPy array` of the system state data with dimension `n` and length `k`.
+    Y
+        The `NumPy array` of the system output data with shape `(m, k)`.
+    U
+        The `NumPy array` of the system control data with shape `(m, k)`.
+    Xdot
+        If not `None`, the `NumPy array` of the system time derivative data with dimension `n` and
+        length `k`. If `None`, `dt` will be used to estimate this data from `X` using
+        an implicit midpoint rule.
+    dt
+        See `Xdot`.
+    H
+        If not `None`, the `NumPy array` representing the systems Hamiltonian.
+    initial_J
+        If not `None`, the `NumPy array` representing the initial
+        guess for the :math:`J \in \mathbb{R}^{(n + m) \times (n + m)}` operator.
+    initial_R
+        If not `None`, the `NumPy array` representing the initial
+        guess for the :math:`R \in \mathbb{R}^{(n + m) \times (n + m)}` operator.
+    initial_alpha
+        The initial value used in calculating the update step.
+    miniter
+        Minimum amount of iterations to perform.
+    maxiter
+        Fail if the iteration count reaches this value without converging.
+    atol
+        Finish when the absolute error measure is below this threshold.
+    rtol
+        Finish when the relative update error measure is below this threshold.
+    weighted_rtol
+        In the weighted pH DMD initialization problem, truncate when the relative value is
+        below this threshold.
+    skew_procrustes_rtol
+        In the skew-symmetric Procrustes problem, truncate when the relative value is
+        below this threshold.
+
+    Returns
+    -------
+    inf_model
+        The inferred |PHLTIModel|.
+    data
+        Dict containing the following fields:
+
+            :update_norms:          |NumPy array| of the norms of the update vectors after
+                                    each iteration.
+            :abs_errs:              |NumPy array| of the absolute errors after each iteration.
+            :rel_errs:              |NumPy array| of the relative  errors after each iteration.
+            :iterations:            Number of total pH DMD iterations.
+            :skew_procrustes_data:  `list` of data generated by the skew-symmetric
+                                    Procrustes subproblems.
+            :weighted_init_data:    If `initial_J` and `initial_R` are both `None`, dict of data
+                                    generated by the weighted pH DMD initialization subproblem.
+            :skew_init_data:        If only `initial_J` is `None`, dict of data generated by the
+                                    skew-symmetric Procrustes subproblem.
+            :spsd_init_data:        If only `initial_R` is `None`, dict of data generated by the psd
+                                    symmetric Procrustes subproblem.
```

Comment:
This should probably be unindented one level. Otherwise, it looks like a quote.

Reply:
For the indentation here I took inspiration from the `Newton`, the `EI`, and the `BFGS` implementations where similar dict key listings are used. In my opinion, these should all follow a similar convention.

Reply:
Right, it would be good to fix all of those.